### PR TITLE
deploy: remove redundant kubernetes-1.18 deployment

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

The only difference between 1.17 and 1.18 were the version numbers:

```
diff -c -r deploy/kubernetes-1.17/test-driver.yaml deploy/kubernetes-1.18/test-driver.yaml
*** deploy/kubernetes-1.17/test-driver.yaml	2020-02-13 17:32:12.156222615 +0100
--- deploy/kubernetes-1.18/test-driver.yaml	2020-10-09 10:46:51.762892764 +0200
***************
*** 1,6 ****
  # This file describes how to test this deployment of the CSI hostpath driver
! # using the Kubernetes 1.17 E2E test suite. For details see:
! # https://github.com/kubernetes/kubernetes/tree/v1.17.0/test/e2e/storage/external

  StorageClass:
    FromName: true
--- 1,6 ----
  # This file describes how to test this deployment of the CSI hostpath driver
! # using the Kubernetes 1.18 E2E test suite. For details see:
! # https://github.com/kubernetes/kubernetes/tree/v1.18.0/test/e2e/storage/external

  StorageClass:
    FromName: true
```

**Special notes for your reviewer**:

The same was already done for 1.19 earlier.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes-latest [hunk 1]
```diff
@@ -1 +1 @@
-kubernetes-1.19
```
Comment (msau42) [inline deploy/kubernetes-latest:1]: Why not leave this at 1.19? If in the future we actually do need to modify 1.19, then we don't have to remember to update latest.

Comment (pohly) [reply deploy/kubernetes-latest:1]: For the sake of consistency (everything links to the actual deployment directory). I can also change it such that it points to 1.19.

Comment (msau42) [reply deploy/kubernetes-latest:1]: Unless there's some problem with symlink chains, I prefer to link to the intention rather than actual.

Comment (pohly) [reply deploy/kubernetes-latest]: Done.

@msau42 : please take another look.

**Comment by msau42** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**